lib/pure/math.nim

```diff
@@ -1182,6 +1182,14 @@ func lcm*[T](x, y: T): T =
 
   x div gcd(x, y) * y
 
+func clamp*[T](val: T, rng: Slice[T]): T {.since: (1, 5).}=
+  ## Like `system.clamp` but takes a slice so you can easily clamp within a range.
+  runnableExamples:
+    doAssert clamp(10, 1 .. 5) == 5
+    doAssert clamp(1, 1 .. 3) == 1
+    doAssert clamp(42.0, 1.0 .. 3.1415926535) == 3.1415926535
+  clamp(val, rng.a, rng.b)
```

Comment:
what's the point of this not raising in debug mode when range is invalid ?
```nim
nim --eval:'echo clamp(10, 5, -1) # -1
nim --eval:'echo clamp(-4, 5, -1) # 5
```
maybe that's a good reason to deprecate system.clamp and make system.clamp correct (using assert)

Reply:
Yea I agree, I'd expect atleast in debug mode that `a <= b` to be checked. Although could do the hand holdy `min(a,b), max(a,b)`, but clearly that's doing more hidden work.

Reply:
I agree. `assert a <= b` should be used.

Reply:
Should I add the assertion in this PR? Should I change anything about `system.clamp` in this PR(deprecate it or anything else)?

Reply:
Doesn't need to be done in this PR IMO.